pypeman/channels.py

```diff
@@ -377,95 +383,218 @@ def handle_and_wait(self, msg):
         """
         return self.loop.run_until_complete(self.handle(msg))
 
+    def _has_callback(self):
+        return hasattr(self, "_callback")
+
+    async def _call_init_nodes(self, msg, start_nodename=None):
+        if self.init_nodes:
+            if start_nodename:
+                node = self.get_node(start_nodename)
+                if node not in self.init_nodes:
+                    raise ValueError("Node named %s not in init nodes", start_nodename)
+            else:
+                node = self.init_nodes[0]
+            msg = await node.handle(msg.copy())
+        return msg
+
+    async def _call_join_nodes(self, msg, start_nodename=None):
+        if self.join_nodes and not self._has_callback():
+            if start_nodename:
+                node = self.get_node(start_nodename)
+                if node not in self.join_nodes:
+                    raise ValueError("Node named %s not in join nodes", start_nodename)
```

Comment:
the word 'named' just makes the message longer.

Perhaps:

`raise ValueError("Node '{start_nodename}' not in join nodes")`

can close

Reply:
done